extensions/easydict/src/google.ts

```diff
@@ -0,0 +1,140 @@
+/*
+ * @author: tisfeng
+ * @createTime: 2022-07-22 23:27
+ * @lastEditor: tisfeng
+ * @lastEditTime: 2022-07-27 22:05
+ * @fileName: google.ts
+ *
+ * Copyright (c) 2022 by tisfeng, All Rights Reserved.
+ */
+
+import axios, { AxiosError, AxiosResponse } from "axios";
+import querystring from "node:querystring";
+import { checkIfPreferredLanguagesContainedChinese } from "./detectLanguage";
+import { RequestErrorInfo, RequestTypeResult, TranslationType } from "./types";
+import { getLanguageItemFromYoudaoId } from "./utils";
+
+export async function requestGoogleTranslate(
+  queryText: string,
+  fromLanguage: string,
+  targetLanguage: string
+): Promise<RequestTypeResult> {
+  console.warn(`---> request google`);
+  // if has preferred Chinese language or ip in China, use cn, else use com.
+  let tld = "com"; // cn,com
+  if (checkIfPreferredLanguagesContainedChinese() || (await checkIfIpInChina())) {
+    tld = "cn";
+  }
+  return googleCrawlerTranslate(queryText, fromLanguage, targetLanguage, tld);
+}
+
+/**
+ *  Check if ip is in China
+ *
+ *  Todo: should store ip in LocalStorage.
+ */
+async function checkIfIpInChina(): Promise<boolean> {
```

Comment:
won't this produce bad results if people are using VPN or are actually outside of their country?

Reply:
@danielsequeira  Also, proxy requests are not currently supported here.  I'm new to js, by the way, is there a way for nodejs to support users using proxies? I mean the ability to have extensions use the system proxy directly without having to manually set it in code.